lib/puppet-lint/optparser.rb

```diff
@@ -108,19 +112,18 @@ def self.build
         opts.on("--no-#{check}-check", "Skip the #{check} check.") do
           PuppetLint.configuration.send("disable_#{check}")
         end
-        unless PuppetLint.configuration.send("#{check}_enabled?")
-          opts.on("--#{check}-check", "Enable the #{check} check.") do
-            PuppetLint.configuration.send("enable_#{check}")
-          end
+
+        next if PuppetLint.configuration.send("#{check}_enabled?")
+
+        opts.on("--#{check}-check", "Enable the #{check} check.") do
+          PuppetLint.configuration.send("enable_#{check}")
         end
       end
 
       opts.load('/etc/puppet-lint.rc')
-      begin
-        opts.load(File.expand_path('~/.puppet-lint.rc')) if ENV['HOME']
-      rescue Errno::EACCES
-        # silently skip loading this file if HOME is set to a directory that
-        # the user doesn't have read access to.
+      if ENV.key?('HOME') && File.readable?(ENV['HOME'])
+        home_dotfile_path = File.expand_path('~/.puppet-lint.rc')
+        opts.load(home_dotfile_path) if File.readable?(home_dotfile_path)
```

Comment:
Reference to an old issue. Although unusual, it's perfectly legitimate to have a user with a home directory that they don't have read access to. OptionParser's load method already checks and silently skips over non-existent files, but doesn't handle cases where the parent directory of that file is unreadable. This behaviour may have changed since this condition was originally placed, and should be reevaluated but it's out of the scope of this PR.